spark/src/main/scala/org/apache/comet/rules/RewriteJoin.scala

```diff
@@ -48,7 +64,7 @@ object RewriteJoin extends JoinSelectionHelper {
 
   def rewrite(plan: SparkPlan): SparkPlan = plan match {
     case smj: SortMergeJoinExec =>
-      getBuildSide(smj.joinType) match {
+      getSmjBuildSide(smj) match {
         case Some(BuildRight) if smj.joinType == LeftSemi =>
           // TODO this was added as a workaround for TPC-DS q14 hanging and needs
           // further investigation
```

Comment:
Not a blocker but just wondering if this this because we were choosing wrong side?

Reply:
This is needed both before and after this PR. I think it may be related to excessive spilling and shuffle but it still needs to be investigated.